akka-actor-typed/src/main/scala/akka/actor/typed/internal/Restarter.scala

```diff
@@ -161,6 +170,17 @@ import akka.actor.typed.scaladsl.Actor
 
   override def loggingEnabled: Boolean = strategy.loggingEnabled
 
+  override def init(ctx: ActorContext[T]) =
+    try {
+      wrap(Behavior.validateAsInitial(Behavior.undefer(behavior, ctx)), afterException = false)
+    } catch {
+      case NonFatal(ex: Thr) ⇒
+        log(ctx, ex)
+        // we haven't actually wrapped and increased retries yet, so need to compare with +1
```

Comment:
when is it incremented?

Reply:
Inside of wrap, but `undefer` is what throws here, so it never is called